integration-tests/test-extension/src/test/java/io/quarkus/it/extension/FinalFieldReflectionInGraalITCase.java

```diff
@@ -4,10 +4,10 @@
 
 import org.junit.jupiter.api.Test;
 
-import io.quarkus.test.junit.SubstrateTest;
+import io.quarkus.test.junit.NativeImageTest;
 import io.restassured.RestAssured;
 
-@SubstrateTest
+@NativeImageTest
 public class FinalFieldReflectionInGraalITCase {
```

Comment:
This test is still annotated with `@SubstrateTest` to make sure we don't break the deprecated classes until their removal. We should probably keep it that way for now and add a comment to explain why it is not annotated with `@NativeImageTest`.

Reply:
Good to know this. 
```
// Use @SubstrateTest to ensure backward compatibility. 
// TODO replace with @NativeImageTest once SubstrateTest is removed. 
```

WDYT?

Reply:
That sounds good!